arches/install/arches-templates/tests/base_test.py

```diff
@@ -0,0 +1,155 @@
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import os
+from contextlib import contextmanager
+
+from django.test import TestCase
+from arches.app.models.graph import Graph
+from arches.app.models.models import Ontology
+from arches.app.models.system_settings import settings
+from arches.app.utils.betterJSONSerializer import JSONSerializer, JSONDeserializer
+from arches.app.utils.data_management.resource_graphs.importer import import_graph as ResourceGraphImporter
+from arches.app.utils.data_management.resources.importer import BusinessDataImporter
+from tests import test_settings
+from arches.app.utils.context_processors import app_settings
+from django.db import connection
+from django.core import management
+from django.test.runner import DiscoverRunner
+
+from arches.app.search.mappings import (
+    prepare_terms_index,
+    delete_terms_index,
+    prepare_concepts_index,
+    delete_concepts_index,
+    prepare_search_index,
+    delete_search_index,
+)
+
+# these tests can be run from the command line via
+# python manage.py test tests --pattern="*.py" --settings="tests.test_settings"
+
+OAUTH_CLIENT_ID = "AAac4uRQSqybRiO6hu7sHT50C4wmDp9fAmsPlCj9"
+OAUTH_CLIENT_SECRET = "7fos0s7qIhFqUmalDI1QiiYj0rAtEdVMY4hYQDQjOxltbRCBW3dIydOeMD4MytDM9ogCPiYFiMBW6o6ye5bMh5dkeU7pg1cH86wF6B\
+        ap9Ke2aaAZaeMPejzafPSj96ID"
+CREATE_TOKEN_SQL = """
+        INSERT INTO public.oauth2_provider_accesstoken(
+            token, expires, scope, application_id, user_id, created, updated)
+            VALUES ('{token}', '1-1-2068', 'read write', 44, {user_id}, '1-1-2018', '1-1-2018');
+    """
+DELETE_TOKEN_SQL = "DELETE FROM public.oauth2_provider_accesstoken WHERE application_id = 44;"
+
+
+class ArchesTestRunner(DiscoverRunner):
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs["debug_mode"] = True
+        # Unless the user has something other than the Django default, give them
+        # what they probably want.
+        if kwargs["pattern"] == "test*.py":
+            kwargs["pattern"] = "*.py"
+        super().__init__(*args, **kwargs)
+
+    def setup_databases(self, **kwargs):
+        ret = super().setup_databases(**kwargs)
+
+        app_settings()  # adds languages to system
+        prepare_terms_index(create=True)
+        prepare_concepts_index(create=True)
+        prepare_search_index(create=True)
+
+        return ret
+
+    def teardown_databases(self, old_config, **kwargs):
+        delete_terms_index()
+        delete_concepts_index()
+        delete_search_index()
+
+        super().teardown_databases(old_config, **kwargs)
+
+class ArchesTestCase(TestCase):
+    def __init__(self, *args, **kwargs):
+        super(ArchesTestCase, self).__init__(*args, **kwargs)
+        if settings.DEFAULT_BOUNDS is None:
+            management.call_command("migrate")
+            with open(os.path.join("tests/fixtures/system_settings/Arches_System_Settings_Model.json"), "r") as f:
```

Comment:
This may already be on your radar, but this path won't exist if your cwd is the project.

```py
  File "/Users/jwalls/newproj/tests/base_test.py", line 94, in __init__
    with open(
         ^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'tests/fixtures/system_settings/Arches_System_Settings_Model.json'
```

We probably don't even need `ArchesTestCase`, it's just here for some screwy workarounds with the testing of oauth in core. [AfS just uses plain TestCase from Django](https://github.com/archesproject/arches-for-science/blob/ba547c16a053d7b82be64455f5c3b8673b24dff8/arches_for_science/tests.py#L4).

What do you think about removing/simplifying this part of the template?